addons/block_code/ui/block_canvas/block_canvas.gd

```diff
@@ -148,6 +154,32 @@ func _drop_obj_property(at_position: Vector2, data: Variant) -> void:
 	reconnect_block.emit(block)
 
 
+func _drop_files(at_position: Vector2, data: Variant) -> void:
+	var resource_files = data["files"]
+	var next_position = at_position
+	const bias = 20
+
+	for file_path in resource_files:
+		# Prepare a Variable block getting the file's resource path.
+		var block_definition = _res_file_block_def(file_path)
```

Comment:
The function name `_res_file_block_def` is not self-explanatory. Maybe `_get_block_definition_from_resource_file()`? Or at least `_get_blockdef_from_resource_file()`?

By the way, that function is using `BlocksCatalog.get_resource_block_definition` which is more clear. Maybe the helper function is not needed at all?

Reply:
This comes from the following implementation of loading audio files, such as "ogg".  However, it is not included in this PR.  To be clear code, yes, using `BlocksCatalog.get_resource_block_definition()` may be a better way now.